Hacker news clone

toaney opened the pull request.

pierrebeaucamp commented on src/pages/news.js:
```diff
@@ -0,0 +1,77 @@
+
+    useEffect(() => {
+        getPage();
+    });
```
what's the reason for using two `useEffect` that call `getPage` in turn?

toaney replied:
Duplicate getPage function call removed

pierrebeaucamp commented on src/scss/components/_header.scss:
```diff
@@ -0,0 +1,71 @@
+
+.header-login-container {
+
+}
```
This empty rule can be removed

toaney replied:
".header-login-container" rule removed

pierrebeaucamp commented on src/scss/layouts/_layout.scss:
```diff
@@ -0,0 +1,21 @@
+body {
+    margin: 0px;
+    font-family: Verdana, Geneva, sans-serif;
```
This conflicts with `src/index.css`

toaney replied:
Unused index.css deleted

pierrebeaucamp commented on src/serviceWorker.js:
```diff
@@ -0,0 +1,137 @@
+// This optional code is used to register a service worker.
+// register() is not called by default.
```
I think this file can also be removed

pierrebeaucamp commented on README.md:
```diff
@@ -1,2 +1,34 @@
+
+Launches the test runner in the interactive watch mode.<br />
+See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+
```
Consider removing this section if there are no unit-test present

toaney replied:
This section will remain since tests for Header, Footer, and Comment components were added.

pierrebeaucamp commented:
Oh, one thing I forgot: Please don't commit / remove the `.DS_Store` files

toaney commented:
Hi Pierre,

Thank you for taking the time to thoroughly review my code and giving such
rich feedback. I'll get to addressing some of the issues mentioned in your
comments.

Do you have a timeline of when you expect issues to be addressed? I have a
few prior commitments today and later this week and want to allot my time
effectively.

Cheers,

Thomas



On Mon, Jan 13, 2020 at 9:20 AM Pierre Beaucamp <notifications@github.com>
wrote:

> Assigned #1
> <https://github.com/toaney/gravitational-code-challenge/pull/1> to @toaney
> <https://github.com/toaney>.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/toaney/gravitational-code-challenge/pull/1?email_source=notifications&email_token=AGCD6XIDAIZBQ6TKV5VAACTQ5SPFNA5CNFSM4KF5E5AKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOV55NKSY#event-2944062795>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGCD6XKOE36J7B5NLDPFAQTQ5SPFNANCNFSM4KF5E5AA>
> .
>

pierrebeaucamp commented:
No rush, take as much time as you need.

toaney commented on README.md:
```diff
@@ -1,2 +1,34 @@
+# Hacker News Clone
+The goal of this project is to recreate a light weight version of hackernews using React Hooks
+
+## Available Scripts
```
"npm install" section added

toaney commented on package.json:
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "hacker-news-clone",
+  "version": "0.1.0",
+  "private": true,
```
"license tag" added

toaney commented on package.json:
```diff
@@ -0,0 +1,39 @@
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
```
"eject" script removed

toaney commented on public/index.html:
```diff
@@ -0,0 +1,43 @@
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Page meta description updated

toaney commented on public/manifest.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
"short_name" and "name" updated

alex-kovoy commented on src/pages/item.js:
```diff
@@ -0,0 +1,73 @@
+    };
+
+    useEffect(() => {
+        const abortController = new AbortController();
```
```jsx
const abortController = new AbortController();
```

I suggest to take a closer look at how this AbortController is supposed to work.

toaney commented:
Hi Alexey,

Thank you for reviewing my code and providing your feedback. I will begin
addressing your comments and reach out if I have any questions.

Regards,
Thomas

On Tue, Jan 14, 2020 at 7:07 PM Alexey Kontsevoy <notifications@github.com>
wrote:

> *@alex-kovoy* approved this pull request.
>
> Congrats on completing the challenge! 👍
> ------------------------------
>
> In src/index.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366671469>
> :
>
> > @@ -0,0 +1,22 @@
>
> +import React from 'react';
>
> +import ReactDOM from 'react-dom';
>
> +import './scss/styles.scss';
>
> +import { BrowserRouter as Router, Route } from 'react-router-dom';
>
> +import News from './pages/news.js';
>
> +import Item from './pages/item.js';
>
> +import Header from './components/header';
>
> +import Footer from './components/footer';
>
> +
>
> +ReactDOM.render(
>
> +    <Router>
>
> +        <div className="page-content">
>
> +            <Header />
>
> +            <Route exact path="/item" component={ Item } />
>
>
> It's better to wrap the routes in statement which would guarantee that
> only 1 route renders.
>
> I would also add a default route to handle 404 errors.
> ------------------------------
>
> In src/pages/item.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366672308>
> :
>
> > @@ -0,0 +1,73 @@
>
> +import React, { useState, useEffect } from 'react';
>
> +import { useParams } from 'react-router';
>
> +import axios from 'axios';
>
> +import Comment from '../components/comment';
>
> +import AddComment from '../components/addComment';
>
> +import getTimestamp from '../components/getTimestamp';
>
> +
>
> +const Item = () => {
>
> +    const [ article, setArticle ] = useState({});
>
> +    const [ comments, setComments ] = useState([]);
>
> +
>
> +    // useParams is a custom hook from react-router that gives access to the url id (page number)
>
> +    let { id } = useParams();
>
>
> I would parse this input parameter and display 404 error if invalid. You
> can write your own function something like:
>
> const pageNumber = usePageNumber()
>
>
> if(pageNumber === -1){
>
>    return <Error404/>
>
> }
>
> Without this basic error handling the app will crash if pageN is not a
> number.
> ------------------------------
>
> In src/pages/news.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366672785>
> :
>
> > +        if (isNaN(currentPage)){
>
> +            return
>
> +        } else {
>
> +            setNextPage(currentPage + 1)
>
> +        }
>
> +    }
>
> +
>
> +    // API call to get intial HN Topstories on page load
>
> +    const get_assets = () => {
>
> +        axios
>
> +        .get("https://hacker-news.firebaseio.com/v0/topstories.json")
>
> +        .then(res => {
>
> +            setArticleList(...[res.data])
>
> +        })
>
> +        .catch((err) => {
>
> +            console.log(err)
>
>
> Kudos for error handling although it's better to display an error to end
> user.
> ------------------------------
>
> In src/pages/news.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366673350>
> :
>
> > +
>
> +    // API call to get intial HN Topstories on page load
>
> +    const get_assets = () => {
>
> +        axios
>
> +        .get("https://hacker-news.firebaseio.com/v0/topstories.json")
>
> +        .then(res => {
>
> +            setArticleList(...[res.data])
>
> +        })
>
> +        .catch((err) => {
>
> +            console.log(err)
>
> +        });
>
> +    };
>
> +
>
> +    // get intial HN Topstories on page load
>
> +    useEffect(() => {
>
> +        const abortController = new AbortController();
>
>
> Looks like you are missing a signal part of it
> <https://developer.mozilla.org/en-US/docs/Web/API/AbortController>.
> ------------------------------
>
> In src/pages/item.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366675465>
> :
>
> > +                <span className="item-stats-item item-link">{getTimestamp(article.time)}</span><span className="article-stats-item">|</span>
>
> +                <span className="item-stats-item item-link">hide</span><span className="article-stats-item">|</span>
>
> +                <span className="item-stats-item item-link">past</span><span className="article-stats-item">|</span>
>
> +                <span className="item-stats-item item-link">web</span><span className="article-stats-item">|</span>
>
> +                <span className="item-stats-item item-link">favorite</span><span className="article-stats-item">|</span>
>
> +                <span className="article-stats-item article-link">{article.kids? article.kids.length : 0} comments</span>
>
>
> To avoid this code duplication I would refactor it to separate components
> something like
>
>     <StateItem> by {article.by}</StateItem>
>
>     <StateItem> {getTimestamp(article.time)}</StateItem><Pipe/>
>
>     <StateItem> hide</StateItem><Pipe/>
>
>     <StateItem> past</StateItem><Pipe/>
>
>     <StateItem> web</StateItem><Pipe/>
>
>     <StateItem> favorite</span><Pipe/>
>
> ------------------------------
>
> In src/pages/news.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366675762>
> :
>
> > @@ -0,0 +1,89 @@
>
> +import React, { useState, useEffect } from 'react';
>
> +import axios from 'axios';
>
> +import Article from '../components/article';
>
> +import { useParams } from 'react-router';
>
> +import { Link } from 'react-router-dom';
>
> +
>
> +const News = () => {
>
> +    const [articleList, setArticleList ] = useState([]);
>
> +    const [ page, setPage ] = useState(1);
>
> +    const [ startIndex, setStartIndex ] = useState(1);
>
> +    const [ nextPage, setNextPage ] = useState("2")
>
> +
>
> +    // useParams is a custom hook from react-router that gives access to the url id (page number)
>
> +    let { id } = useParams();
>
>
> app will crash if id is not a number.
> ------------------------------
>
> In src/pages/item.js
> <https://github.com/toaney/gravitational-code-challenge/pull/1#discussion_r366677169>
> :
>
> > +
>
> +    // API call to get article on initial page load
>
> +    const getArticle = () => {
>
> +        axios
>
> +        .get(`https://hacker-news.firebaseio.com/v0/item/${ id }.json`)
>
> +        .then(res => {
>
> +            setArticle(res.data)
>
> +            setComments(res.data.kids)
>
> +        })
>
> +        .catch((err) => {
>
> +            console.log(err)
>
> +        });
>
> +    };
>
> +
>
> +    useEffect(() => {
>
> +        const abortController = new AbortController();
>
>
> const abortController = new AbortController();
>
> I suggest to take a closer look at how this AbortController is supposed to
> work.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/toaney/gravitational-code-challenge/pull/1?email_source=notifications&email_token=AGCD6XNZ5BIDSHG4ADYTE4LQ5Z4WNA5CNFSM4KF5E5AKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCRYSK2Y#pullrequestreview-342959467>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGCD6XKI5SZCHEFTC4ZIQA3Q5Z4WNANCNFSM4KF5E5AA>
> .
>
